src/uct/tcp/tcp_ep.c

```diff
@@ -322,12 +271,16 @@ UCS_CLASS_DEFINE_NAMED_DELETE_FUNC(uct_tcp_ep_destroy_internal,
 
 void uct_tcp_ep_destroy(uct_ep_h tl_ep)
 {
-    uct_tcp_ep_t *ep = ucs_derived_of(tl_ep, uct_tcp_ep_t);
+    uct_tcp_ep_t *ep       = ucs_derived_of(tl_ep, uct_tcp_ep_t);
+    uct_tcp_iface_t *iface = ucs_derived_of(ep->super.super.iface,
+                                            uct_tcp_iface_t);
 
     if ((ep->conn_state == UCT_TCP_EP_CONN_STATE_CONNECTED) &&
         ucs_test_all_flags(ep->flags, UCT_TCP_EP_CTX_CAPS)) {
         /* remove TX capability, but still will be able to receive data */
+        uct_tcp_cm_remove_ep(iface, ep, 1);
         uct_tcp_ep_remove_ctx_cap(ep, UCT_TCP_EP_FLAG_CTX_TYPE_TX);
+        uct_tcp_cm_insert_ep(iface, ep);
```

Comment:
why need to remove and then add?

Reply:
> why need to remove and then add?

now we don't do insert into iface's EP list inside `uct_tcp_cm_remove_ep()` and remove from iface's EP list inside `uct_tcp_cm_remove_ep()`